src/Dofs/ConstraintHandler.jl

```diff
@@ -154,6 +158,34 @@ function _add!(ch::ConstraintHandler, dbc::Dirichlet, interpolation::Interpolati
     append!(ch.prescribed_dofs, constrained_dofs)
 end
 
+function add_node_dbc!(ch::ConstraintHandler, dbc::Dirichlet, interpolation::Interpolation, field_dim::Int, offset::Int)
+    if interpolation !== default_interpolation(getcelltype(ch.dh.grid))
+        warn("adding constraint to nodeset is not recommended for sub/super-parametric approximations.")
+    end
+    constrained_dofs = Int[]
+    _celldofs = fill(0, ndofs_per_cell(ch.dh))
+    added_nodes = Set{Int}() # keep track of the nodes we have added to
+    for (cellidx, cell) in enumerate(ch.dh.grid.cells)
+        idx = findfirst(x -> in(x, dbc.faces) && !in(x, added_nodes), cell.nodes)
```

Comment:
The time complexity of this part is O(ncells * length(dbc.faces)), it is basically a masked nested loop. There is also one hidden assumption if I understand correctly which is that every cell can at most result in a single node being found/added. So when 2 boundary nodes belong to 1 and only 1 cell, 1 of them will be missed. Think of 2 nodes on the edge of a corner quadratic hexahedron. I know that quadratic hexahedrons are not supported but there is probably a better way to do this?

Reply:
I think `find` instead of `findfirst` will solve the second problem.

Reply:
> There is also one hidden assumption if I understand correctly which is that every cell can at most result in a single node being found/added. 

Yea, good catch, will fix this.

TBH I don't think the time for this will matter at all compared to other things, but I can do some benchmarks.